### PR TITLE
PN-11746 - Handle multi-language on PG

### PR DESCRIPTION
**Pull request by alessandrogelmi**
## Short description
Handle multi-language on PG and keep the selected language on refresh page.
This PR resolve: [PN-11751](https://pagopa.atlassian.net/browse/PN-11751), [PN-11752](https://pagopa.atlassian.net/browse/PN-11752) and [PN-11754](https://pagopa.atlassian.net/browse/PN-11754)

## List of changes proposed in this pull request
- Get language from token-exchange URLs from Selfcare
- Store lang on sessionStorage
- Add lang params to external URLs

## How to test
On PG, use a direct access link and append a lang query parameter with a language (e.g. `&lang=en`) to it. Verify that the page is translated into the selected language. Also, try refreshing the page and check that the selected language is retained

[PN-11751]: https://pagopa.atlassian.net/browse/PN-11751?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PN-11752]: https://pagopa.atlassian.net/browse/PN-11752?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PN-11754]: https://pagopa.atlassian.net/browse/PN-11754?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/pn-personagiuridica-webapp/src/navigation/SessionGuard.tsx [hunk 1]
```diff
@@ -151,6 +155,13 @@ const SessionGuard = () => {
     return params.get('#selfCareToken');
   }, [location]);
 
+  const handleSetUserLanguage = useCallback(() => {
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/navigation/SessionGuard.tsx:158]: Try to move this code into the App.tsx file

File: packages/pn-personagiuridica-webapp/src/navigation/__test__/SessionGuard.test.tsx [hunk 2]
```diff
@@ -23,6 +23,7 @@ vi.mock('react-router-dom', async () => ({
 vi.mock('react-i18next', () => ({
   useTranslation: () => ({
     t: (str: string) => str,
+    i18n: { changeLanguage: () => new Promise(() => {}) },
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/navigation/__test__/SessionGuard.test.tsx:26]: Remove this

File: packages/pn-commons/src/utility/multilanguage.utility.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import * as yup from 'yup';
+
+import { LangCode } from '@pagopa/mui-italia';
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/utility/multilanguage.utility.ts:3]: Replace the use of LangCode with string

File: packages/pn-personagiuridica-webapp/src/App.tsx [hunk 4]
```diff
@@ -24,7 +24,7 @@ import {
   useMultiEvent,
   useTracking,
 } from '@pagopa-pn/pn-commons';
-import { PartyEntity, ProductEntity } from '@pagopa/mui-italia';
+import { LangCode, PartyEntity, ProductEntity } from '@pagopa/mui-italia';
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/App.tsx:27]: Replace the use of LangCode with string

File: packages/pn-personagiuridica-webapp/src/App.tsx [hunk 5]
```diff
@@ -97,7 +98,7 @@ const ActualApp = () => {
       {
         id: '1',
         title: t('header.product.organization-dashboard'),
-        productUrl: routes.PROFILE(organization?.id),
+        productUrl: routes.PROFILE(organization?.id, currentLanguage),
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/App.tsx:101]: use i18n.language directly here and below

File: packages/pn-personagiuridica-webapp/src/navigation/routes.const.ts [hunk 6]
```diff
@@ -3,6 +3,7 @@ import {
   PRIVACY_LINK_RELATIVE_PATH as PRIVACY_POLICY,
   TOS_LINK_RELATIVE_PATH as TERMS_OF_SERVICE,
 } from '@pagopa-pn/pn-commons';
+import { LangCode } from '@pagopa/mui-italia';
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/navigation/routes.const.ts:6]: Replace the use of LangCode with string

File: packages/pn-personagiuridica-webapp/src/pages/Contacts.page.tsx [hunk 7]
```diff
@@ -18,10 +19,11 @@ import { useAppDispatch, useAppSelector } from '../redux/hooks';
 import { RootState } from '../redux/store';
 
 const Contacts = () => {
-  const { t } = useTranslation(['recapiti']);
+  const { t, i18n } = useTranslation(['recapiti']);
   const dispatch = useAppDispatch();
+  const currentLanguage = i18n.language as LangCode;
   const organization = useAppSelector((state: RootState) => state.userState.user.organization);
-  const profileUrl = PROFILE(organization?.id);
+  const profileUrl = PROFILE(organization?.id, currentLanguage);
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/pages/Contacts.page.tsx:26]: use i18n.language directly

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

File: packages/pn-personagiuridica-webapp/src/navigation/__test__/SessionGuard.test.tsx (hunk 2 above)
Comment (alessandrogelmi) [reply packages/pn-personagiuridica-webapp/src/navigation/__test__/SessionGuard.test.tsx:26]: If I remove this lines the test throws an error:
`TypeError: Cannot read properties of undefined (reading 'changeLanguage')`

Comment (alessandrogelmi) [reply packages/pn-personagiuridica-webapp/src/navigation/__test__/SessionGuard.test.tsx]: Moving `handleSetUserLanguage` to `App.tsx` resolved the error.

**Review by AndreaCimini90 — APPROVED** [review]

**Pull request merged**